test/files/neg/t8079a.scala

```diff
@@ -0,0 +1,4 @@
+trait C[-I] {
+  private[this] type X = C[I]
+  def f2(b: X): Unit
```

Comment:
- [ ] Investigate why this doesn't give a variance error (even before this PR) when `X` is `protected[this]`.